modules/develop/pages/data-transforms/versioning-compatibility.adoc

```diff
@@ -0,0 +1,67 @@
+= Versioning and Compatibility for Data Transforms
+:description: The data transforms SDKs use semantic versioning to ensure compatibility and stability. Use this guide to learn the SDKs that are compatible with different versions of Redpanda, and what guarantees are provided regarding SDK and Redpanda compatibility.
+
+{description}
+
+== Semantic versioning
+
+The data transforms SDKs use semantic versioning, which is a versioning scheme with a three-part number: `major.minor.patch`. When a new version is released, one part of the version number is changed. The changed part of the version number signifies the level of change:
+
+* *Major*: Indicate breaking changes.
+* *Minor*: Indicate backward-compatible functionality.
+* *Patch*: Indicate backward-compatible bug fixes.
+
+For example, in the version number `1.2.3`, `1` is the major version, `2` is the minor version, and `3` is the patch version.
+
+== Compatibility matrix
+
+This compatibility matrix provides detailed information on which versions of the SDK are supported with which versions of Redpanda. Always consult the compatibility matrix to ensure that you are using compatible versions.
+
+=== Golang SDK
+
+|===
+| Golang SDK Version | Redpanda Version
+
+| 1.0.x       | 24.1.x, 23.3.x
+| 0.2.x       | 23.3.x
+| 0.1.x       | 23.3.x
+|===
+
+=== Rust SDK
+
+|===
+| Rust SDK Version | Redpanda Version
+
+| 1.0.x       | 24.1.x, 23.3.x
+| 0.2.x       | 23.3.x
+| 0.1.x       | 23.3.x
+|===
+
+
+== Backward compatibility
+
+SDK versions will continue to work in the supported versions of Redpanda. However, they may not work in newer versions of Redpanda. Always check the compatibility matrix when upgrading either the SDK or Redpanda.
```

Comment:
> However, they may not work in newer versions of Redpanda.

These two sentences to me counter each other.

Maybe something like:

SDK versions will have backwards compatibilty and continuted to be supported in newer versions of the broker. However, newer SDK versions may not work with older versions of Redpanda. Always check this compatibility matrix when upgrading the SDK or Redpanda.